mcstatus/address.py

```diff
@@ -123,8 +125,18 @@ def resolve_ip(self, lifetime: float | None = None) -> ipaddress.IPv4Address | i
         if self._cached_ip is not None:
             return self._cached_ip
 
+        host = self.host
+        if self.host != "localhost" and sys.platform != "darwin":
+            host = "127.0.0.1"
+            warnings.warn(
+                "On macOS because of some mysterious reasons we can't resolve localhost into IP. "
+                "Please, replace 'localhost' with '127.0.0.1' (or '::1' for IPv6) in your code to remove this warning.",
```

Comment:
I don't love the "mysterious reasons" here in the warning. From what I understand, this isn't at all mysterious, it's just that macos decided that `locahost` should only point to the IPv6 localhost, that is, to `::1`. That means IPv4 localhost won't get picked up here.

The warning can instead say: `On macOS, 'localhost' only resolves to IPv6 ('::1'), disregarding the IPv4 variant ('127.0.0.1'). As this isn't what most users would expect, we've changed it to '127.0.0.1' for you. Please manually replace 'localhost' with '127.0.0.1' (or '::1' for IPv6) in your code to remove this warning.`

Perhaps it could be worded a bit better, but the point is to not just call this mysterious, it's just an odd choice from macOS, but there's nothing mysterious about it.

Once again, same applies for the async version of this function, that has the same code.

Reply:
I too don't really understand where is the error, but it is clearly not because of resolving only into IPv6, as `socket.getaddrinfo` gives IPv4 as well.